src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -136,6 +147,65 @@ def _store_area_penalties(self, results_dict, area):
         results_dict["totals_with_penalties"] = \
             results_dict["total_cost"] + results_dict["penalty_cost"]
 
+    @property
+    def cumulative_bills(self):
+        return {
+            uuid: {
+                "name": results["name"],
+                "spent_total": round_floats_for_ui(results['spent_total']),
+                "earned": round_floats_for_ui(results['earned']),
+                "penalties": round_floats_for_ui(results['penalties']),
+                "total": round_floats_for_ui(results['total'])
+            }
+            for uuid, results in self.cumulative_bills_results.items()
+        }
+
+    def update_cumulative_bills(self, area):
+        for child in area.children:
+            self.update_cumulative_bills(child)
+
+        if area.uuid not in self.cumulative_bills_results or \
+                ConstSettings.GeneralSettings.KEEP_PAST_MARKETS is True:
```

Comment:
I think this check could be skipped if only last_past_market data is accumulated.

Reply:
Of course it can work even without it, if we access only the last past market. However, this defeats the purpose of always recalculating it when the KEEP_PAST_MARKETS is set to true.